### DISCUSSION TITLE
Is there a better way to observe the "in-flight" status?

### DISCUSSION THREAD

**Discussion by Czajnikowski** [pointfreeco/swift-composable-architecture#1424] [Ideas]
In one of the sample code files, we use [`isNumberFactRequestInFlight` boolean](https://github.com/pointfreeco/swift-composable-architecture/blob/bef2084cb26263f08a959aa00b127edfbc2df267/Examples/CaseStudies/SwiftUICaseStudies/02-Effects-Basics.swift#L26) to keep an eye on the status of the network request. The problem with that is that it's actually a secondary source of truth that has to be manually maintained, and the ultimate source of such status is in the effect itself (?). I'm wondering if there is a better way to go about such status.

I likely missed some episode that fixes this issue, or if that's not the case then maybe we can think about an improvement?

**Comment by stephencelis** [comment]
@Czajnikowski There’s no way to query for an effect’s status directly, but you could use the lifecycle of `await viewStore.send` to manage some local state during the call:

```swift
@State var isNumberFactRequestInFlight = false

var body: some View {
  // …
  Button("Number fact") {
    self.isNumberFactRequestInFlight = true
    Task {
      await viewStore.send(.numberFactButtonTapped).finish()
      self.isNumberFactRequestInFlight = false
    }
  }
}
```

You lose some testability here, but not much, and arguably surface area that isn’t worth testing at the domain layer.

If you have other ideas of how to solve the problem, please do share!
